File: sandbox/demo/fs.py
from pathlib import Path


def read_text(path: Path | str) -> str:
    try:
        return Path(path).read_text(encoding="utf-8")
    except FileNotFoundError:
        return "<missing>"
    except PermissionError as e:
        return f"<permission denied: {e}>"
    except OSError as e:
        return f"<error: {e}>"


def write_text(path: Path | str, data: str) -> None:
    Path(path).write_text(data, encoding="utf-8")

File: sandbox/demo/fs/fs.go
package fs

import (
	"errors"
	"fmt"
	"os"
)

func ReadText(path string) string {
	data, err := os.ReadFile(path)
	if err != nil {
		if errors.Is(err, os.ErrNotExist) {
			return "<missing>"
		}
		if errors.Is(err, os.ErrPermission) {
			return fmt.Sprintf("<permission denied: %v>", err)
		}
		return fmt.Sprintf("<error: %v>", err)
	}
	return string(data)
}

func WriteText(path string, data string) error {
	return os.WriteFile(path, []byte(data), 0644)
}
